unittests/test_package_generator.py

```diff
@@ -0,0 +1,86 @@
+import venv
+from pathlib import Path
+
+import pytest
+
+from pypi2nix.logger import Logger
+from pypi2nix.pip.virtualenv import VirtualenvPip
+from pypi2nix.requirement_parser import RequirementParser
+from pypi2nix.requirement_set import RequirementSet
+from pypi2nix.target_platform import TargetPlatform
+
+from .package_generator import PackageGenerator
+
+
+def test_can_generate_two_packages(package_generator: PackageGenerator):
+    package_generator.generate_setuptools_package(name="package1",)
+    package_generator.generate_setuptools_package(name="package2",)
+
+
+def test_can_gerate_source_distribution_with_correct_name(
+    package_generator: PackageGenerator,
+):
+    distribution = package_generator.generate_setuptools_package(name="testpackage")
+    assert distribution.name == "testpackage"
+
+
+def test_can_install_generated_packages(
+    pip,
+    current_platform: TargetPlatform,
+    requirement_parser: RequirementParser,
+    target_directory: Path,
+    install_target: Path,
+    package_generator: PackageGenerator,
+):
+    package_generator.generate_setuptools_package(name="testpackage")
+    requirements = RequirementSet(current_platform)
+    requirements.add(requirement_parser.parse("testpackage"))
+    pip.install(
+        requirements,
+        source_directories=[str(target_directory)],
+        target_directory=str(install_target),
+    )
```

Comment:
memo: Assert that the packages were actually installed.  Using pip.freeze() would probably be a good approach.